keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
-+ _COSI_ - Container _Object_ Store Interface, modeled after CSI.
-+ _cosi-node-adapter_ - a pod per node which receives Kubelet gRPC _NodeStageVolume_ and _NodeUnstageVolume_ requests, ensures the target bucket has been provisioned, and notifies the kubelet when the pod can be run.
-+ _driver_ - a container (usually paired with a sidecar container) that is responsible for communicating with the underlying object store to create, delete, grant access to, and revoke access from buckets. Drivers talk gRPC and need no knowledge of Kubernetes. Drivers are typically written by storage vendors, and should not be given any access outside of their namespace.
-+ _driverless_ - a use-case where no driver exists to support the backend object store. Some COSI automation may still be provided, but backend operations such as creating a bucket or minting credentials will not occur.
-+ _greenfield bucket_ - a new backend bucket created by COSI.
-+ _green-to-brownfield_ - a use-case where COSI creates a new bucket on behalf of a user, and then supports ways for other users in the cluster to share this bucket.
-+ _provisioner_ - a generic term meant to describe the combination of a sidecar and driver. "Provisioning" a bucket can mean creating a new bucket or granting access to an existing bucket.
-+ _sidecar_ - a Kubernetes-aware container (usually paired with a driver) that is responsible for fullfilling COSI requests with the driver via gRPC calls. The sidecar's access can be restricted to the namespace where it resides, which is expected to be the same namespace as the provisioner. The COSI sidecar is provided by the Kubernetes community.
-+ _static provisioning_ - custom resource creation is done by the admin rather than via COSI automation. This may also include _driverless_ but they are independent.
+#### Functionality
 
-# Proposal
++ Support automated **Bucket Creation**
++ Support automated **Bucket Deletion**
++ Support automated **Access Credential Generation**
++ Support automated **Access Credential Revokation**
++ Support automated **Bucket Provisioning** for workloads (enabling pods to access buckets)
++ Support **Bucket Reuse** 
 
-## APIs
+#### System Properties
 
-### Storage APIs
++ Support **workload portability** across clusters
++ Achieve the above goals in a **vendor neutral** manner
++ Standardize mechanism for third-party vendors to integrate easily with COSI
 
-#### BucketRequest
+### Non-Goals
 
-A user facing, namespaced custom resource requesting provisioning of a new bucket, or requesting access to an existing bucket. A `BucketRequest` (BR) lives in the app's namespace.  In addition to a `BucketRequest`, a [BucketAccessRequest](#bucketaccessrequest) is necessary in order to grant credentials to access the bucket. BRs are required for both greenfield and brownfield uses.
+**Data Plane** API standardization will not be addressed by this project
```

Comment:
Can add a non-goal of _not_ being cluster orchestrator agnostic?

Reply:
I think that is not needed as this is a KEP in Kubernetes.
If it is a design doc for COSI gRPC interfaces, then we can mention that.